### PR TITLE
fixed check for lev zero module creating

### PR DESCRIPTION
**Pull request by makart19**

### CODE REVIEW

File: cpp/daal/include/services/internal/sycl/level_zero_module_sycl.h [hunk 1]
```diff
@@ -68,8 +68,14 @@ class ZeModule : public Base
...
     {
-        const auto ptr = new ZeModule(deviceQueue, binarySize, pBinary, status);
-        if (!ptr) status |= ErrorMemoryAllocationFailed;
+        auto ptr = new ZeModule(deviceQueue, binarySize, pBinary, status);
+        if (!status)
+        {
+            if (ptr) delete ptr;
+            ptr = nullptr;
+        }
+        else if (!ptr)
```
Comment (amgrigoriev) [inline cpp/daal/include/services/internal/sycl/level_zero_module_sycl.h:77]: Maybe it makes sense to check "ptr" before "status"? Just in order to ensure that ErrorMemoryAllocationFailed is added in any case.

**Review by amgrigoriev — APPROVED** [review]

File: cpp/daal/include/services/internal/sycl/level_zero_module_sycl.h (hunk 1 above)
Comment (makart19) [reply cpp/daal/include/services/internal/sycl/level_zero_module_sycl.h:77]: It is more probable to get incorrect status than nullptr and provides more info, therefore i set it first.

**Pull request merged**